tests/page_test.go

```diff
@@ -636,63 +630,117 @@ func TestPageWaitForFunction(t *testing.T) {
-			let resp = page.waitForFunction(%s, %s, %s);
+		const test = async function() {
+			let resp = await page.waitForFunction(%s, %s, %s);
 			if (resp) {
 				log('ok: '+resp.innerHTML());
 			} else {
 				log('err: '+err);
-			}`
+			}
+		}`
 
 		s := fmt.Sprintf(script, `"document.querySelector('h1')"`, "{ polling: 100, timeout: 2000, }", "null")
-		_, err := tb.runJavaScript(s)
+		err = runJSInEventLoop(t, vu, s)
 		require.NoError(t, err)
-		assert.Contains(t, log, "ok: Hello")
+		assert.Contains(t, *log, "ok: Hello")
 	})
 
 	t.Run("ok_func_poll_mutation", func(t *testing.T) {
 		t.Parallel()
 
-		tb := newTestBrowser(t)
-		p := tb.NewPage(nil)
-		require.NoError(t, tb.runtime().Set("page", p))
-		var log []string
-		require.NoError(t, tb.runtime().Set("log", func(s string) { log = append(log, s) }))
+		vu, rt, log, cleanUp := startIteration(t)
+		defer cleanUp()
 
-		_, err := tb.runJavaScript(`fn = () => document.querySelector('h1') !== null`)
+		_, err := rt.RunString(`fn = () => document.querySelector('h1') !== null`)
 		require.NoError(t, err)
 
-		p.Evaluate(tb.toGojaValue(`() => {
+		_, err = rt.RunString(`
+		const page = browser.newPage();
+		page.evaluate(() => {
 			console.log('calling setTimeout...');
 			setTimeout(() => {
 				console.log('creating element...');
 				const el = document.createElement('h1');
 				el.innerHTML = 'Hello';
 				document.body.appendChild(el);
 			}, 1000);
-		}`))
+		})`)
+		require.NoError(t, err)
+
+		script := `
+		const test = async function() {
+			let resp = await page.waitForFunction(%s, %s, %s);
+			log('ok: '+resp);
+		}`
 
 		s := fmt.Sprintf(script, "fn", "{ polling: 'mutation', timeout: 2000, }", "null")
-		_, err = tb.runJavaScript(s)
+		err = runJSInEventLoop(t, vu, s)
 		require.NoError(t, err)
-		assert.Contains(t, log, "ok: null")
+		assert.Contains(t, *log, "ok: null")
 	})
 }
 
+// startIteration will work with the event system to start chrome and
+// (more importantly) set browser as the mapped browser instance which will
+// force all tests that work with this to go through the mapping layer.
+// This returns a cleanup function which should be deferred.
+func startIteration(t *testing.T) (*k6test.VU, *goja.Runtime, *[]string, func()) {
+	t.Helper()
+
+	vu := k6test.NewVU(t)
+	rt := vu.Runtime()
+
+	mod := browser.New().NewModuleInstance(vu)
+	jsMod, ok := mod.Exports().Default.(*browser.JSModule)
+	require.Truef(t, ok, "unexpected default mod export type %T", mod.Exports().Default)
+
+	// Setting the mapped browser into the vu's goja runtime.
+	require.NoError(t, rt.Set("browser", jsMod.Browser))
+
+	// Setting log, which is used by the callers to assert that certain actions
+	// have been made.
+	var log []string
+	require.NoError(t, rt.Set("log", func(s string) { log = append(log, s) }))
+
+	vu.ActivateVU()
+	vu.StartIteration(t)
+
+	return vu, rt, &log, func() { t.Helper(); vu.EndIteration(t) }
+}
+
+func runJSInEventLoop(t *testing.T, vu *k6test.VU, js string) error {
```

Comment:
I would recommend using [modulestest](https://pkg.go.dev/go.k6.io/k6@v0.48.0/js/modulestest)

Reply:
We're using `modulestest`, what specifically from that package?

Reply:
Got it to work, thanks for the heads up. b244cf026fc7dea00505ac185829b90d21133db9